src/executables/unload_object.c

```diff
@@ -661,13 +669,24 @@ extract_objects (const char *exec_name, const char *output_dirname, const char *
 
 	  if (!datafile_per_class && (!required_class_only || IS_CLASS_REQUESTED (i)))
 	    {
-	      SPLIT_USER_SPECIFIED_NAME (sm_ch_name ((MOBJ) class_ptr), owner_name, class_name);
-	      if (text_print
-		  (obj_out, NULL, 0, "%cid %s%s%s.%s%s%s %d\n", '%', PRINT_IDENTIFIER (owner_name),
-		   PRINT_IDENTIFIER (class_name), i) != NO_ERROR)
+	      if (ctxt.is_dba_user || ctxt.is_dba_group_member)
```

Comment:
Even I'm not sure it is effieient but I'll propose a solution to simplify, it up to you its just idea:
1. define a function like:
```
char *
print_owner_name (char * owner_name, bool multi_schema)
{
  char buf[DB_MAX_USER_LENGTH+4] = "";

  if (multi_schema)
    {
      sprintf (buf, "%s%s%s%s", PRINT_IDENTIFIER (owner_name), ".")
    }

  return buf;
}
```
2. Next re-write following if-else-block as follow:
```
if (text_print (obj_out, NULL, 0, "%cid %s%s%s%s %d\n", '%', print_owner_name (class_name, ctxt.is_dba_user || ctxt.is_dba_group_member), PRINT_IDENTIFIER (class_name), ) != NO_ERROR)
  {
    status = 1;
    goto end;
  }
```